test/ConfigManagerTest.cpp

```diff
@@ -1180,18 +1180,17 @@ That is, instead of returning `bool`, it now returns
 template <typename... ValidatorParameterTypes>
 auto transformNonExceptionValidatorIntoExceptionValidator(
     Validator<ValidatorParameterTypes...> auto validatorFunction,
-    std::string_view errorMessage) {
+    std::string errorMessage) {
   // The whole 'transformation' is simply a wrapper.
-  return
-      [validatorFunction, errorMessage = std::string(std::move(errorMessage))](
-          const ValidatorParameterTypes&... args)
-          -> std::optional<ErrorMessage> {
-        if (std::invoke(validatorFunction, args...)) {
-          return std::nullopt;
-        } else {
-          return std::make_optional<ErrorMessage>(errorMessage);
-        }
-      };
+  return [validatorFunction, errorMessage = std::move(errorMessage)](
+             const ValidatorParameterTypes&... args)
+             -> std::optional<ErrorMessage> {
+    if (std::invoke(validatorFunction, args...)) {
+      return std::nullopt;
+    } else {
+      return std::make_optional<ErrorMessage>(errorMessage);
+    }
+  };
```

Comment:
Why do you need this helper function here again? don't you already have it in the `ConfigManager.h`?

Reply:
Yes, but I don't have access to it from the tests. In short, the friend declaration for tests doesn't work with free standing functions and I'm `99%` sure, that a friend function, which is only actually defined in the test case, could lead to linker errors.

Reply:
As for why I need it: In order to make the testing with validators easier, I created a function for generating validators with a specific invariant. When I later wrote the tests for `ExceptionValidator`s I, of course, wanted to re-use as **much** as possible.

So, instead of polluting things with another generator function, which just does the same thing slightly different, I just wrote a transformer function.

If wanted, I can write an expanded explanation. But be warned, the infrastructure for testing validators get a tad hard to understand, in order to make the testing as exhausting as possible.